$NetBSD: patch-configure.ac,v 1.3 2025/02/08 02:58:55 taca Exp $

* Adjust PHP directories.
* Adjust PHP library name.
* Don't automatically add libgcc on SunOS. Reported upstream as #75941.

--- configure.ac.orig	2022-10-31 10:36:04.000000000 +0000
+++ configure.ac
@@ -117,12 +117,14 @@ PHP_RELEASE_VERSION=[$]3
 PHP_EXTRA_VERSION=[$]4
 PHP_VERSION="$PHP_MAJOR_VERSION.$PHP_MINOR_VERSION.$PHP_RELEASE_VERSION$PHP_EXTRA_VERSION"
 PHP_VERSION_ID=`expr [$]PHP_MAJOR_VERSION \* 10000 + [$]PHP_MINOR_VERSION \* 100 + [$]PHP_RELEASE_VERSION`
+PHP_VER=[$]1[$]2
 
 dnl Allow version values to be used in Makefile.
 PHP_SUBST(PHP_MAJOR_VERSION)
 PHP_SUBST(PHP_MINOR_VERSION)
 PHP_SUBST(PHP_RELEASE_VERSION)
 PHP_SUBST(PHP_EXTRA_VERSION)
+PHP_SUBST(PHP_VER)
 
 dnl Define where extension directories are located in the configure context.
 AC_DEFUN([PHP_EXT_BUILDDIR],[$config_m4_dir])dnl
@@ -264,9 +266,6 @@ AX_CHECK_COMPILE_FLAG([-fvisibility=hidd
 case $host_alias in
   *solaris*)
     CPPFLAGS="$CPPFLAGS -D_POSIX_PTHREAD_SEMANTICS"
-    if test "${enable_libgcc+set}" != "set" && test "$GCC" = "yes"; then
-      enable_libgcc=yes
-    fi
     ;;
   *dgux*)
     CPPFLAGS="$CPPFLAGS -D_BSD_TIMEOFDAY_FLAVOR"
@@ -323,9 +322,9 @@ dnl SAPI configuration.
 dnl ----------------------------------------------------------------------------
 
 dnl Paths to the targets are relative to the build directory.
-SAPI_SHARED=libs/libphp[]$PHP_MAJOR_VERSION[.]$SHLIB_DL_SUFFIX_NAME
-SAPI_STATIC=libs/libphp[]$PHP_MAJOR_VERSION[.a]
-SAPI_LIBTOOL=libphp[]$PHP_MAJOR_VERSION[.la]
+SAPI_SHARED=libs/libphp[]$PHP_VER[.]$SHLIB_DL_SUFFIX_NAME
+SAPI_STATIC=libs/libphp[]$PHP_VER[.a]
+SAPI_LIBTOOL=libphp[]$PHP_VERS[.la]
 
 PHP_CONFIGURE_PART(Configuring SAPI modules)
 
@@ -1212,12 +1211,12 @@ test "$program_suffix" = "NONE" && progr
 
 case $libdir in
   '${exec_prefix}/lib')
-    libdir=$libdir/php
+    libdir=$libdir/php/${PHP_API_VERS}
     ;;
 esac
 case `eval echo $datadir` in
   '${prefix}/share')
-    datadir=$datadir/php
+    datadir=$datadir/php/${PHP_API_VERS}
     ;;
 esac
 
@@ -1267,7 +1266,7 @@ case $PHP_LAYOUT in
     datarootdir=$prefix/share
     ;;
   *)
-    datarootdir=$prefix/php
+    datarootdir=$prefix/${PHP_SHAREDIR}
     ;;
 esac
 
@@ -1394,6 +1393,7 @@ PHP_SUBST(PHP_RPATHS)
 PHP_SUBST(PHP_SAPI)
 PHP_SUBST_OLD(PHP_VERSION)
 PHP_SUBST_OLD(PHP_VERSION_ID)
+PHP_SUBST_OLD(PHP_VER)
 PHP_SUBST(SHELL)
 PHP_SUBST(SHARED_LIBTOOL)
 PHP_SUBST(PHP_FRAMEWORKS)
@@ -1469,7 +1469,7 @@ PHP_SUBST(all_targets)
 PHP_SUBST(install_targets)
 PHP_SUBST(install_binary_targets)
 
-PHP_INSTALL_HEADERS([Zend/ TSRM/ include/ main/ main/streams/])
+PHP_INSTALL_HEADERS([Zend/ TSRM/ main/ main/streams/])
 
 PHP_ADD_SOURCES(TSRM, TSRM.c, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
 
